database/postgresql/src/main/java/com/example/javaopencommerce/catalog/PsqlCategoryRepositoryImpl.java

```diff
@@ -0,0 +1,32 @@
+package com.example.javaopencommerce.catalog;
+
+import com.example.javaopencommerce.exception.EntityNotFoundException;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+
+
+class PsqlCategoryRepositoryImpl implements PsqlCategoryRepository {
+
+  private final EntityManager em;
```

Comment:
Why not extending a `JpaRepository` and move those Native Queries to Entities, with `@Query` annotation?

Reply:
Hmm more control? Dunno, we can create ticket for that if u want